script/setup_heroku

```diff
@@ -1,45 +1,49 @@
-#!/bin/sh
-
-set -e
+#!/bin/bash
+
+set -ue
+
+function create_heroku_addon () {
+	if heroku addons:info "$1" &> /dev/null; then
+		echo "Already installed $1"
+	else
+		echo "Installing $1"
+		heroku addons:create "$@"
+	fi
+}
+
+if [ $# -eq 0 ]; then
+	echo >&2 "Please add your Heroku app name to the end of the command."
+	echo >&2 "Usage: setup_heroku your_app_name"
+	exit 1
+fi
 
-if [ $1 ]
-then
-  herokuApp="$1"
+export HEROKU_APP="$1"
 
-  echo "Getting ready to set environment variables for $herokuApp"
-  figaro heroku:set -e production --app $herokuApp
+echo "Getting ready to set environment variables for $HEROKU_APP"
+figaro heroku:set -e production
 
-  echo "Setting MAILER_URL"
-  heroku config:set MAILER_URL=$herokuApp.herokuapp.com --app $herokuApp
+echo "Setting MAILER_URL"
+heroku config:set MAILER_URL="$HEROKU_APP.herokuapp.com"
 
-  echo "Setting DEVISE_SECRET_KEY"
-  # generate a random string with 36 characters
-  token1=$(python -c 'import uuid; print (uuid.uuid4())')
-  heroku config:set DEVISE_SECRET_KEY=$token1 --app $herokuApp
+if [ -z "$(heroku config:get SECRET_TOKEN)" ]; then
+	echo "Setting SECRET_TOKEN"
+	# generate a random string with 36 characters
+	heroku config:set SECRET_TOKEN="$(ruby -rsecurerandom -e 'puts SecureRandom.uuid')"
```

Comment:
I think those two goals are contradictory. If you don't require a recent Ruby to be installed locally, then you can't run `rake secret` locally either. And I don't think you can do `heroku run rake secret` until the first push is done since Heroku won't know the buildpack and `rake` won't work remotely. If I'm wrong about the latter, then you could use `heroku run rake secret` before doing `git push heroku master`, but that would have to be tested with a blank Heroku app.

I also don't think you can expect Python to be installed out of the box on Windows, but I could be wrong there too. Then again, perhaps if you've got `bash` installed (which is required to have a shell with `$(...)` support), then you may have `python` too.

As for `securerandom`, it seems to exist all the way back to Ruby 1.8, so anyone who has Ruby installed should be able to execute that command. It seems reasonable to me that someone wanting to work on a Ruby application should be able to have a working installation of a modern Ruby.

I suppose another option would be to run these local commands with `docker` since that seems to be your preferred local development option out of the box, but I think that opens its own can of worms.